frontend/app/src/App.tsx

```diff
@@ -1061,6 +1061,11 @@ export class App extends PureComponent<Props, State> {
         newDialog,
         sessionEvent.scriptCompilationException?.message ?? "No message"
       )
+    } else if (sessionEvent.type === "scriptChangedOnDisk") {
+      // If runOnSave is enabled, automatically trigger a rerun with fresh context info
+      if (this.state.userSettings.runOnSave) {
```

Comment:
nit: I think we can combine this into one condition: `sessionEvent.type === "scriptChangedOnDisk" && this.state.userSettings.runOnSave`